Stream errors

bukka opened the pull request:
This introduces new stream error handling that allows configurable handling of stream errors. This can be configured in stream contex.

RFC: https://wiki.php.net/rfc/stream_errors

DanielEScherzer reviewed (approved):
Updating `ext/reflection/tests/ReflectionExtension_getClassNames_basic.phpt` for new classes looks good to me, okay for ext/reflection

bukka committed:
Add new stream erros API

This introduces new stream error handling that allows configurable
handling of stream errors. This can be configured in stream contex.

All streams, ext/standard and ext/phar stream errors are converted to
use the new API.

RFC: https://wiki.php.net/rfc/stream_errors

Closes GH-20524

jrfnl commented:
Quick question: the RFC for this PR is still listed under ["Pending Implementation / Landing"](https://wiki.php.net/rfc#pending_implementationlanding). Is there anything still missing ? Or should this RFC be moved into the "Implemented in PHP 8.6" list ?
